docs/native.md

```diff
@@ -0,0 +1,236 @@
+
+**`altis.analytics.data.metrics <array>`**
+
+Similar to the `altis.analytics.data.attributes` filter but allows you to pass metrics based on server-side data.
+
+**`altis.analytics.data <array>`**
+
+Filters the entire array passed to the client side.
+
+**`altis.analytics.elasticsearch.url <string>`**
+
+Filters the Elasticsearch server URL.
+
+### Functions
+
+**`Altis\Analytics\Utils\query( array $query, array $params = [] ) : array`**
+
+Queries the analytics data with the Elasticsearch Query DSL provided in `$query`. `$params` will be added as query string parameters to the Elasticsearch request URL.
+
+**`Altis\Analytics\Utils\get_elasticsearch_url() : string`**
+
+Get the Elasticsearch server URL.
+
+**`Altis\Analytics\Utils\milliseconds() : integer`**
+
+Get the current time since the unix epoch in milliseconds.
+
+**`Altis\Analytics\Utils\merge_aggregations( array $current, array $new, string $bucket_type = '' ) : array`**
+
+A utility function for merging aggregations returned by Elasticsearch queries. This is necessary to keep a store of your analytics data as Elasticsearch indexes can be rotated or lost.
+
+## Querying Data
+
+Data can be queried from Elasticsearch, providing a powerful tool for filtering and aggregating records. If you are using Altis requests are automatically signed.
+
+It is recommended to use the `Altis\Analytics\Utils\query()` function for this.
+
+### Anatomy of an event record
+
+A user session covers every event recorded between opening the website and closing it. For every event recorded the following data is recorded depending on the scope.
+
+- `event_type`: The type of event recorded, eg. "pageView", "click", "_session.start" or "_session.stop".
+- `event_timestamp`: The timestamp in milliseconds of when the event was recorded on the site.
+- `attributes`: String data associated with event.
+  - `session`: Unique ID across all page views.
+  - `pageSession`: Unique ID for one page view.
+  - `url`: The current page URL.
+  - `hash`: The current URL hash.
+  - `referer`: The page referer.
+  - `utm_campaign`: The Urchin Tracker campaign from the query string if set.
+  - `utm_source`: The Urchin Tracker source from the query string if set.
+  - `utm_medium`: The Urchin Tracker medium from the query string if set.
+  - Any attributes added via the `altis.analytics.data.attributes` filter.
+  - Any attributes added via `Altis.Analytics.registerAttribute()` or passed to `Altis.Analytics.record()`.
+- `metrics`: Numeric data associated with an event.
+  - `scrollDepthMax`: Maximum scroll depth on page so far. Percentage value between 1-100.
+  - `scrollDepthNow`: Scroll depth at time of event. Percentage value between 1-100.
+  - `elapsed`: Time elapsed in milliseconds since the start of the page view.
+  - Any metrics added via `Altis.Analytics.registerMetric()` or passed to `Altis.Analytics.record()`.
+- `endpoint`
```

Comment:
As I understand it, and `endpoint` is unique per user, but what is a "user". If I log in to my phone and computer, do I use the same endpoint? If I am logged out, and then log in, is that using the same endpoint?

If I update values `enpdoint.Demographic.Locale`, is that just setting those value for this event, or will it also set the endpoint ("user") data for subsequent records (i.e. the endpoint/user data is a separate, single object?)

Reply:
Soz - an endpoint is a device & browser. They're only connected via `endpoint.User.UserId` if set, eg. if you log in on multiple browsers / devices

Reply:
Oh and yeah - whatever data is associated with the endpoint will be automatically merged in with each event record, so they're handled separately.

Reply:
I added a new "Key Concepts" section that answers your questions 👍